gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/login/LoginHideFormHandler.java

```diff
@@ -47,37 +50,59 @@ public LoginHideFormHandler(Domain domain) {
+                .orElse(false);
+        boolean hasExactlyOneSocialProvider = socialProviders != null && socialProviders.size() == 1;
 
-        // hide form option disabled, continue
-        if (!isHideForm) {
-            routingContext.next();
-            return;
-        }
+        boolean shouldSkipLoginForm = isHideFormEnabled && hasExactlyOneSocialProvider;
+        boolean hasError = !routingContext.queryParam(ConstantKeys.ERROR_PARAM_KEY).isEmpty();
 
-        // no external provider, continue
-        if (socialProviders == null) {
-            routingContext.next();
-            return;
-        }
 
-        // more than one external provider, continue
-        if (socialProviders.size() != 1) {
+        // hide form option disabled, continue
+        if (!shouldSkipLoginForm) {
             routingContext.next();
             return;
         }
 
-        doRedirect(routingContext, socialProviders.get(0));
+        if (hasError) {
+            redirectWithError(routingContext);
+        } else {
+            redirectToProvider(routingContext, socialProviders.get(0));
+        }
     }
 
-    private void doRedirect(RoutingContext routingContext, IdentityProvider identityProvider) {
+    private void redirectToProvider(RoutingContext routingContext, IdentityProvider identityProvider) {
         Map<String, String> urls = routingContext.get(SOCIAL_AUTHORIZE_URL_CONTEXT_KEY);
         String redirectUrl = urls.get(identityProvider.getId());
         routingContext.response()
                 .putHeader(io.vertx.core.http.HttpHeaders.LOCATION, redirectUrl)
                 .setStatusCode(302)
                 .end();
     }
+
+    private void redirectWithError(RoutingContext context) {
+        /* AM-3381: normally if we're at /login with an error, the form displays it.
+               However, when the login form is hidden, we don't have a place to display it, and redirecting to the provider
+               may cause a redirect loop. */
+
+        var error = context.queryParam(ERROR_PARAM_KEY).stream().findFirst().orElse("");
+        var errorCode = context.queryParam(ERROR_CODE_PARAM_KEY).stream().findFirst().orElse("");
+        var errorDescription = context.queryParam(ERROR_DESCRIPTION_PARAM_KEY).stream().findFirst().orElse("");
+
+        var redirectUri = context.request().getParam(Parameters.REDIRECT_URI);
+        if (StringUtils.isBlank(redirectUri)) {
+            context.fail(500, new IllegalStateException("Received an error but there's nowhere to redirect to. error=%s, error_code=%s, error_description=%s".formatted(error, errorCode, errorDescription)));
+            return;
+        }
+        var targetUri = UriBuilder.fromURIString(redirectUri)
+                .addParameter(ConstantKeys.ERROR_PARAM_KEY, error)
+                .addParameter(ERROR_CODE_PARAM_KEY, errorCode)
+                .addParameter(ERROR_DESCRIPTION_PARAM_KEY, UriBuilder.encodeURIComponent(errorDescription))
```

Comment:
**question**: according to lines 89-91, errors parameters can be empty. Do we have to add a parameter if it is empty? (maybe the addParameter ignore it in this case but I'm not sure)

Reply:
fixed